### ISSUE TITLE
Missing `six` module in Python environment in NVDA 2026.1 RC1

### ISSUE THREAD

**Issue by tseykovets** [nvaccess/nvda#20037]
### Brief summary

Missing `six` module in Python environment in NVDA 2026.1 RC1

In NVDA 2025.3.3, the `six` module is available by default in the Python environment used for add-on development. For example, running `import six` in the Python console works without issues.

However, in NVDA 2026.1 RC1, the same statement results in an error:

```
ModuleNotFoundError: No module named 'six'
```

This suggests that the `six` module has been removed or is no longer bundled with NVDA's Python environment. There is no mention of this change in the official documentation or release notes, and no deprecation warning was provided in earlier versions.

The last mention of module six is ​​in the release notes of NVDA 2019.3:

> Updated six to version 1.12.0. (#9630)

Since many existing add-ons may still rely on `six` for Python 2/3 compatibility (even if transitional), its sudden absence can break development workflows or cause runtime failures during testing.

Could you please clarify:
- Was the removal of `six` intentional?
- If yes, is there an official replacement or recommended migration path?
- If not, should it be restored in the bundled environment?

Additionally, it would be helpful to update the documentation or migration guide to reflect this change if it's permanent.

Thank you.

### Steps to reproduce

1. Open the Python console in NVDA 2026.1 RC1.
2. Run the code `import six`

### Actual behavior

console output: `ModuleNotFoundError: No module named 'six'`

### Expected behavior

Successful module import.

### NVDA type

portable copy

### NVDA version

2026.1 RC1

### Have you tried any other versions of NVDA? If so, please report their behaviors.

The problem is not reproducible in NVDA 2025.3.3.

### Windows version

Windows 11 (version 10.0.26100, build 26100)

### Does the issue still occur after restarting your computer?

I have restarted my computer and the issue still occurs

### If NVDA add-ons are disabled, is your problem still occurring?

I have restarted NVDA with add-ons disabled and the issue still occurs

### Does the issue still occur after you run the System Accessibility Repair Tool (COM Registration Fixing Tool in older versions) in NVDA's tools menu?

I have run the System Accessibility Repair Tool and the issue still occurs

**Comment by seanbudd** [comment]
Removing six was intended.
Please see [our developer guide](https://github.com/nvaccess/nvda/blob/master/projectDocs/dev/developerGuide/developerGuide.md#definition-of-the-public-api) for reference on the stability of dependency inclusion.

>Pip packages: These may be updated, downgraded or removed at any time. It is recommended to package any pip dependency directly with your add-on, rather than using NVDA's version of the package. The exception to this is wxPython, which must be used from NVDA.

**Comment by CyrilleB79** [comment]
Shouldn't this removal have been documented though?

I think that other package removal have been advertised in the past: typing_extensions (removed twice by the way!), nose.

And since package upgrade are mentioned in the changes, even if not part of the API, it would make sense to mention removal as well.

Since 2026.1 translation freeze has ended, we cannot add a new translated item in the change log.
We may either add a new item which will remain untranslated for 2026.1, or just do not mention the removal for this time.

Also while at it, clarify if typing_extensions was really removed twice (for 2024.1 and 2026.1). If it wasn't removed in 2026.1 dev cycle, it's still time to remove the dedicated 2026.1 change log item.

Though, it would be nice, for the future, to clarify if package upgrade or removal should be advertised.

**Comment by SaschaCowley** [comment]
@CyrilleB79 

> Shouldn't this removal have been documented though?

In hindsight, yes. If I recall correctly my reasoning was that Six was not only a Pip package (and thus not part of the API), but only included transitively.

> And since package upgrade are mentioned in the changes, even if not part of the API, it would make sense to mention removal as well.

I agree. I'll look into clarifying our documentation around this.

> Since 2026.1 translation freeze has ended, we cannot add a new translated item in the change log. We may either add a new item which will remain untranslated for 2026.1, or just do not mention the removal for this time.

We're going to keep things as-is. I will post to the API list though.

> Also while at it, clarify if typing_extensions was really removed twice (for 2024.1 and 2026.1). If it wasn't removed in 2026.1 dev cycle, it's still time to remove the dedicated 2026.1 change log item.

This is correct. typing_extensions was removed in 2024.1 (#15544 [`requirements.txt`:28](https://github.com/nvaccess/nvda/pull/15544/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552L26-L28)), reintroduced in 2024.2 (#15967 [`requirements.txt`:17](https://github.com/nvaccess/nvda/pull/15967/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552R16-R17)), and re-removed in 2026.1 (#18689 [`pyproject.toml`:37](https://github.com/nvaccess/nvda/pull/18689/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711L36-L37)).

> Though, it would be nice, for the future, to clarify if package upgrade or removal should be advertised.

Agreed.
